packages/emitter-framework/src/typescript/components/type-alias-declaration.tsx

```diff
@@ -1,23 +1,29 @@
 import { refkey as getRefkey } from "@alloy-js/core";
 import * as ts from "@alloy-js/typescript";
-import { Scalar } from "@typespec/compiler";
+import { Type } from "@typespec/compiler";
 import { $ } from "@typespec/compiler/experimental/typekit";
 import { reportDiagnostic } from "../../lib.js";
 import { TypeExpression } from "./type-expression.jsx";
 
 export interface TypedAliasDeclarationProps extends Omit<ts.TypeDeclarationProps, "name"> {
-  type: Scalar;
+  type: Type;
   name?: string;
 }
 
 export type TypeAliasDeclarationProps = TypedAliasDeclarationProps | ts.TypeDeclarationProps;
 
+/**
+ * Create a TypeScript type alias declaration. Pass the `type` alias to emit the
+ * type as the provided TypeSpec type.
+ */
 export function TypeAliasDeclaration(props: TypeAliasDeclarationProps) {
   if (!isTypedAliasDeclarationProps(props)) {
     return <ts.TypeDeclaration {...props}>{props.children}</ts.TypeDeclaration>;
   }
 
-  const originalName = props.name ?? props.type.name;
+  const originalName =
```

Comment:
is that kinda broken if there there is no name still?

Reply:
It behaves the same as before yeah.